### PR TITLE
Feat/today specific date

### PR DESCRIPTION
**Pull request by SirTristam**
Add an option in the Dashboard plugin to use either '>today' or the current date when scheduling tasks for today. 

- Using '>today' will cause the task to automatically roll to the next day if not completed.
- Using the current date will cause the task to remain due on that date if not completed, resulting in it showing as overdue.

There are groups of users who prefer each of these; adding the option to select between the two will make the plugin more useful for those who do not want their overdue tasks to automatically roll to the next day.

### CODE REVIEW

**Comment by dwertheimer** [comment]
1) First of all, as a user who prefers real dates to >today, I am 100% with you. 
2) MORE IMPORTANTLY, thank you so much for making your suggestion as a PR. The only issue is that @jgclark is in the middle of a massive refactor so I don't know how quick this will be to integrate
3) I am happy to test. Can you make some suggestions on the best way to test your code in practice? (I can guess, but it's best if you say so explicitly)

**Comment by dwertheimer** [comment]
One thing I noticed is that you wrote v1.1.3, because v1.1.2 is the latest in the repo, but it looks like v1.2.0 is the latest live version. So @jgclark I think you might need to push the 1.2 code so that @SirTristam can merge in your latest live code and bump the version

FYI The tests are failing -- @jgclark and I are working on that separately

**Comment by dwertheimer** [comment]
@SirTristam I noticed that you have focused on the "Today" button in the popup, but there are other places that things are rescheduled to >today

Screen Cap 2024-04-16 at 14 54 48@2x

Screen Cap 2024-04-16 at 14 55 05@2x

I would suggest, when you get the latest code from @jgclark, you search for ">today" in the whole plugin. You will see other spots that need to be addressed.

Screen Cap 2024-04-16 at 14 56 02@2x

Pls keep track and submit a list of all the variants to test when you submit the revised PR. Thanks!

**Comment by SirTristam** [comment]
@dwertheimer Okay, I'll take a look for the additional places where we move notes to "today" and get them changed too, and verify that the changes are still complete against 1.2.0.

Am I better off enumerating the UI locations where this function is performed in a comment, or editing the PR description? Not sure where this would be most expected.

**Comment by SirTristam** [comment]
It looks like the v1.2.0 code is in the *dashboard-react* branch, so I'll base off of that for now.

**Comment by dwertheimer** [comment]
Comment in the or description is good

**Comment by dwertheimer** [comment]
Yes I am working on getting 1.2 over to main

File: jgclark.Dashboard/CHANGELOG.md
Comment (jgclark) [inline jgclark.Dashboard/CHANGELOG.md:1]: Sorry that I pushed the 1.2 code to the wrong branch, so you could only see v1.1.2.
I'm updating to make this v1.2.1.

**Review by jgclark — COMMENTED** [review]
I join my thanks to @dwertheimer's, for this PR.
I've reviewed and made 2 direct commits to change the version number.
Otherwise, this is all good, and once you've had a chance to update based on @dwertheimer's suggestion to cover other places the same idea applies, I will merge and release.

**Comment by SirTristam** [comment]
> I've reviewed and made 2 direct commits to change the version number.
> 
> Otherwise, this is all good, and once you've had a chance to update based on @dwertheimer's suggestion to cover other places the same idea applies, I will merge and release.

Thanks; I will refresh and review tonight and should have an update tomorrow morning.

**Comment by SirTristam** [comment]
Dashboard plugin uses ">today" to schedule tasks in three locations. Changes have been made to support all three of these locations, instead of just the initial one. Verified that these changes work with both tasks and checklists items.

Precondition for all testing: In the Dashboard settings, the box "Reschedule items in place, rather than move them?" is checked.

Testing: Proper handling of the Dashboard setting item "Use 'today' to schedule tasks for today?"

Locations where these scheduling changes are visible are:
1. In the 'Yesterday' section when it appears. Precondition: An open task was scheduled for yesterday. Activating: Press the 'All -> Today' button. Expected: All tasks and checklist items in the 'Overdue' section will be moved to the 'Today' section, with either '>today' or '>{today's date}', depending upon the configuration setting.
2. In the 'Overdue' section when it appears. Precondition: An open task was scheduled in the past. Activating: Press the 'All -> Today' button. Expected: All tasks and checklist items in the 'Overdue' section will be moved to the 'Today' section, with either '>today' or '>{today's date}', depending upon the configuration setting.
3. In the scheduling popup, displayed by pressing the pencil icon after a task or checklist item. Precondition: A task or checklist item exists on the Dashboard in a section other than 'Today'. Activating: Press the 'Today' button on the scheduling popup for the item. Expectation: The task or checklist item will be moved to the 'Today' section, and examiniation of the task or checklist item will show that the task or checklist item has been scheduled with either '>today' or '>{today's date}', depending upon the configuration setting.

**Comment by dwertheimer** [comment]
Thanks again @SirTristam I just tested this and it works for me. I like this addition!
@jgclark over to you

**Pull request merged**